app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraph.kt

```diff
@@ -11,8 +11,14 @@ data class ControlFlowGraph(
     val conditionalTrueLinks: ReferenceMap<IFTNode, IFTNode>,
     val conditionalFalseLinks: ReferenceMap<IFTNode, IFTNode>
 ) {
-    val finalTreeRoots: List<IFTNode> get() = treeRoots.filter {
-        it !in unconditionalLinks && it !in conditionalTrueLinks && it !in conditionalFalseLinks
+    val finalTreeRoots: List<Pair<IFTNode, CFGLinkType>> get() = treeRoots.filter {
```

Comment:
IntelliJ says that this could be private. Also the name is no longer valid, maybe. `outLinks`, or sth would be better.

Reply:
What I meant is that now `finalTreeRoots` is not a collection of nodes but some more complex structure.